### PR TITLE
feat: Expand analyzer rules to cover template strings

### PR DESCRIPTION
**Pull request by DrBarnabus**
Updates the Roslyn Analyzer with new diagnostics to report more errors in near real-time in the IDE.

This commit adds support for detecting and reporting the following diagnostics;
- COMPOSITE0005 - Empty or invalid template string
- COMPOSITE0006 - Primary key separator missing from template string

## Summary by CodeRabbit

- New Features
  - Added analyser that validates template strings and primary key separators with precise, location-aware diagnostics.
- Bug Fixes
  - Prevents multiple primary separators and improves handling of empty/whitespace templates, malformed braces, and invalid property references.
- Refactor
  - Centralised template validation across components for consistent behaviour and clearer error reporting.
- Tests
  - Added comprehensive test coverage for tokenisation, structure rules, separator handling, diagnostics, and edge cases.
- Chores
  - Removed unused imports and aligned internal namespaces.

### CODE REVIEW

File: src/CompositeKey.Analyzers.Common.UnitTests/Validation/TemplateValidationTests.cs [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+            // Act
+            var result = TemplateValidation.ValidateTemplateString(templateString);
+
+            // Assert
+            result.IsSuccess.ShouldBeFalse();
+            result.Descriptor.ShouldBe(DiagnosticDescriptors.EmptyOrInvalidTemplateString);
+            result.MessageArgs.ShouldNotBeNull();
+            result.MessageArgs.Length.ShouldBe(1);
+            result.MessageArgs[0].ShouldBe(templateString ?? string.Empty);
+        }
```
[the remark this answers is not in the document]
Comment (DrBarnabus) [reply src/CompositeKey.Analyzers.Common.UnitTests/Validation/TemplateValidationTests.cs:42]: The using statements are in a global usings file for the tests, and as such aren't required in each file.
